internal/federationin/federationin.go

```diff
@@ -189,13 +189,17 @@ func (h *handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	timeoutContext, cancel := context.WithTimeout(ctx, h.config.Timeout)
 	defer cancel()
 
-	deps := pullDependencies{
-		fetch:               client.Fetch,
-		insertExposures:     h.publishdb.InsertExposures,
-		startFederationSync: h.db.StartFederationInSync,
+	opts := pullOptions{
```

Comment:
Would it be more efficient if these were all pointers?

Reply:
I think this is OK as is.

the pullDependencies are function pointers

the query is already a pointer

and the whole pullOptions is passed to functions by reference, not by value..